src/Processors/Executors/PipelineExecutor.cpp

```diff
@@ -735,7 +731,16 @@ void PipelineExecutor::executeImpl(size_t num_threads)
                             CurrentThread::detachQueryIfNotDetached();
```

Comment:
It's not easy to understand the difference between these `execute*` functions.
Probably it's better to rename `executeImpl()` into something like `executeMultipleThreads()`?

Reply:
Agree.
Will fix it in other pr.